vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "MapBufferBuilder.h"
+
+using namespace facebook::react;
+
+namespace facebook {
+namespace react {
+
+MapBufferBuilder::MapBufferBuilder()
+    : MapBufferBuilder::MapBufferBuilder(INITIAL_KEY_VALUE_SIZE) {}
+
+MapBuffer MapBufferBuilder::EMPTY() {
+  static auto emptyMap = MapBufferBuilder().build();
+  return emptyMap;
+}
+
+MapBufferBuilder::MapBufferBuilder(uint16_t initialSize) {
+  keyValuesSize_ = initialSize;
+  keyValues_ = new Byte[keyValuesSize_];
```

Comment:
@dannyvv The MapBuffer class has a bunch of potential issues, and the most we can do is submit PRs changing it - otherwise we take it 'as is' from core. (btw what's a memory excursion? There's some custom resize/allocation code in there, if that's what it means.)

I'll remove this file once the core PR is merged and we ingest it.